components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIMgtAdminServiceImpl.java

```diff
@@ -50,15 +54,46 @@ public APIMgtAdminServiceImpl(APISubscriptionDAO apiSubscriptionDAO, PolicyDAO p
     }
 
     /**
-     * @see org.wso2.carbon.apimgt.core.api.APIMgtAdminService#getAPISubscriptions(int)
+     * @see APIMgtAdminService#getPoliciesByLevel(PolicyLevel)
+     */
+    @Override
+    public List<Policy> getPoliciesByLevel(PolicyLevel policyLevel) throws APIManagementException {
+        try {
+            return policyDAO.getPoliciesByLevel(policyLevel);
+
+        } catch (APIMgtDAOException e) {
+            String errorMessage = "Couldn't retrieve Throttle Policies with level: " + policyLevel.name();
+            log.error(errorMessage, e);
+            throw new APIManagementException(errorMessage, e, e.getErrorHandler());
```

Comment:
This is an enhancement that has been done. Error Handler is being set at DAO layer and the existing Error Handler is being passed  to the APIManagementException. This was done to be able to accurately detect difference between for example, searching with an invalid UUID vs actual SQLExceptions. If you set the Error Handler at the impl layer for this case we cant tell if the reason for the failure is a DB error or the use of an invalid UUID.